src/rhsm/utils.py

```diff
@@ -282,3 +286,180 @@ def wrapper(*args, **kwargs):
+
+
+class StatusMessage:
+    """Class for temporary reporting.
+
+    While you can call 'print()' and 'clean()' methods directly, the easier
+    way is to use context manager: 'with StatusMessage("Fetching data"):'.
+
+    This object will print the description to stdout when called. When the
+    context manager exists (either because the code finished or because an
+    error got raised inside), printed output will be cleared before the
+    program continues with its execution. This ensures that the message will
+    disappear after it's no longer valid.
+    """
+
+    def __init__(self, description: Optional[str]):
+        if description is None:
+            description = "Transmitting data"
+        self.raw_text = description
+
+        CURSIVE = "\033[3m"
+        RESET = "\033[0m"
+
+        self.text = f"{CURSIVE}{self.raw_text}...{RESET}"
+
+        self.quiet = False
+        config = rhsm.config.get_config_parser()
+        if config.get("rhsm", "quiet") == "1":
+            self.quiet = True
+        if not sys.stdout.isatty():
+            self.quiet = True
+
+    def print(self):
+        if self.quiet:
+            return
+        print(self.text, end="\r")
+
+    def clean(self):
+        if self.quiet:
+            return
+        print(" " * len(self.text), end="\r")
+
+    def __enter__(self):
+        self.print()
+
+    def __exit__(self, error_type, error_value, traceback):
+        self.clean()
+        if error_type:
+            raise
+
+
+class StatusSpinnerStyle:
+    LINE: List[str] = ["|", "/", "-", "\\"]
+    BRAILLE: List[str] = ["⠋", "⠙", "⠸", "⠴", "⠦", "⠇"]
+    WIDE_BRAILLE: List[str] = ["⠧ ", "⠏ ", "⠋⠁", "⠉⠉", "⠈⠙", " ⠹", " ⠼", "⠠⠴", "⠤⠤", "⠦⠄"]
+    BAR_FORWARD: List[str] = ["[    ]", "[=   ]", "[==  ]", "[=== ]", "[====]", "[ ===]", "[  ==]", "[   =]"]
+    BAR_BACKWARD: List[str] = ["[    ]", "[   =]", "[  ==]", "[ ===]", "[====]", "[=== ]", "[==  ]", "[=   ]"]
+    BAR_BOUNCE: List[str] = BAR_FORWARD + BAR_BACKWARD
+
+
+class LiveStatusMessage(StatusMessage):
+    """Class for temporary reporting, with activity spinner.
+
+    While you can call 'print()' and 'clean()' methods directly, the easier
+    way is to use context manager: 'with LiveStatusMessage("Fetching data"):'.
+
+    This object will print the description to stdout when called. When the
+    context manager exists (either because the code finished or because an
+    error got raised inside), printed output will be cleared before the
+    program continues with its execution. This ensures that the message will
+    disappear after it's no longer valid.
+
+    You can set several loading styles (via class StatusSpinnerStyle),
+    choose if the spinner should be placed before or after the description
+    and choose the speed in which the animation is played (per frame, in ms).
+    """
+
+    def __init__(
+        self,
+        description: Optional[str],
+        *,
+        style: List[str] = StatusSpinnerStyle.BRAILLE,
```

Comment:
I would use `StatusSpinnerStyle.LINE` as default spinner. When I tried to use on Non-gui TTY (e.g. try to use it with ctrl-alt-F3), then I just didn't see braille characters and the spinner just doesn't indicate any progress.

We can try to be tricky a little. When environment variable `$TERM` is `xterm-256color` and not only `linux`, then change the spinner to `BRAILLE` and maybe use italic font. It would require more testing with more terminals (KDE terminal, xterm, lxterminal, etc.)

Reply:
I'd keep things simple, and use only one spinner type. Otherwise we need to test the different spinners separately, with the proper environment/terminal for it, etc.

Reply:
Done, using the LINE spinner.